### ISSUE TITLE
Error when testing telegraf config on Windows: driver "sqlite" not supported

### ISSUE THREAD

**Issue by kuzyn** [influxdata/telegraf#11259]
### Relevant telegraf.conf

```toml
[[inputs.sql]]
  interval = "5m"
  driver = "sqlite"
  dsn = "C:/PATH/TO/DB"

  [[inputs.sql.query]]
    query="SELECT * FROM tableName ORDER BY day DESC LIMIT 1"
```

### Logs from Telegraf

```text
[telegraf] Error running agent: could not initialize input inputs.sql: driver "sqlite" not supported use one of [clickhouse cockroach maria mssql mysql pgx postgres snowflake sqlserver tidb]
```

### System info

Telegraf 1.22.4; Windows 10 Pro

### Steps to reproduce

1. Configure `conf` to point to sqlite database
2. Run telegraf with the `--test` and `--debug` switches 
3. It gives the error above
...

### Expected behavior

Works and show a test example of the sql metrics

### Actual behavior

Gives the error aboves

### Additional info

This config works on Linux (with updated dsn path). The database can be loaded via the `sqlite3.exe` util, so I know it's not corrupt... Is there a limitation with telegraf, sqlite and Windows 10?

**Comment by powersj** [comment]
It is/was a limitation of the sqlite library itself that Telegraf users. Telegraf imports [modernc.org/sqlite](https://pkg.go.dev/modernc.org/sqlite) as a driver for sqlite. 

If you look at our [import file](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/sql/drivers_sqlite.go), it is only enabled for linux, freebsd, and darwin. Looking at the sqlite driver's project, it was not until last November that Windows support was fully enabled:

> 2021-11-13 v1.14.0: 
> Support windows/amd64. This target had previously only experimental status because of a now resolved memory leak. 

I have created https://github.com/influxdata/telegraf/pull/11260 which updates the library and adds the windows build flags. Once artifacts are built, can you please try testing those?

Thanks!

**Comment by kuzyn** [comment]
Hi Joshua, thanks for triggering that quick build! Same error on [telegraf-1.23.0~acea120f_windows_amd64.zip](https://output.circle-artifacts.com/output/job/d81d7b63-76b0-4deb-a2f3-0a5751b50607/artifacts/0/build/dist/telegraf-1.23.0~acea120f_windows_amd64.zip)

Going from the structure in `inputs/plugins/sql` and the `drivers_sqlite.go` and `drivers.go` files... is it possible that it's not getting pulled into the available driver list in the same way as other drivers?

https://github.com/powersj/telegraf/blob/fix/11259/plugins/inputs/sql/sql.go#L329

**Comment by powersj** [comment]
@kuzyn thanks for the quick attempt. I have pushed another few changes and confirmed that I get past the missing driver error and get to connect to the database failed. Can you try again with the latest artifacts?

Thanks!

**Comment by kuzyn** [comment]
Yup @powersj works fine on my end as well. Thanks for the fix! Do you expect this to be rolled into 1.23 ?

**Comment by powersj** [comment]
> Yup @powersj works fine on my end as well. Thanks for the fix! Do you expect this to be rolled into 1.23 ?

yes - I just pushed a formatting change for go.mod, so I'll need a review from someone on the team tomorrow, and then I should be able to push and release in next week's v1.23 release!

Thanks for trying it out for me!
